### PR TITLE
NC first Commit

### PR DESCRIPTION
**Pull request by nurseda916**
Guys, can you review it if you have a chance

### CODE REVIEW

File: src/test/java/com/library/steps/B28G31_US190_StepDefs_NC.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    String endPoint;
+    int int1;
+    String contentType;
+    RequestSpecification request;
+    @Given("I logged Library api as a {string}-NC")
+    public void i_logged_library_api_as_a_nc(String userType) {
+    request = given().log().uri()
+                .header("x-library-token", LibraryAPI_Util.getToken(userType));
+
+             this.userType = userType;
```
Comment (farukgoc) [inline src/test/java/com/library/steps/B28G31_US190_StepDefs_NC.java:26]: Why do you need this? You dont use that userType anywhere else

File: src/test/java/com/library/steps/B28G31_US190_StepDefs_NC.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        this.headerType = headerType;
+
+        request.accept(headerType);
+
+    }
+    @When("I send GET request to {string} endpoint-NC")
+    public void i_send_get_request_to_endpoint_nc(String endPoint) {
+
+                this.endPoint = endPoint;
```
Comment (farukgoc) [inline src/test/java/com/library/steps/B28G31_US190_StepDefs_NC.java:41]: You dont need this as well. The other endpoints you will use on the following steps could be different

File: src/test/java/com/library/steps/B28G31_US190_StepDefs_NC.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+                      request.accept(headerType)
+                       .when().get(ConfigurationReader.getProperty("library.baseUri") + endPoint);
```
Comment (farukgoc) [inline src/test/java/com/library/steps/B28G31_US190_StepDefs_NC.java:44]: You could have assign this to a variable to use for following steps to have cleaner code.

File: src/test/java/com/library/steps/B28G31_US190_StepDefs_NC.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    @Then("{string} field should not be null-NC")
+    public void field_should_not_be_null_nc(String field) {
+
+        if (field.equalsIgnoreCase("id")) {
+
+            request.accept(contentType)
+                    .when().get(ConfigurationReader.getProperty("library.baseUri") + endPoint)
+                    .then().statusCode(int1)
+                    .contentType(contentType)
+                    .body("id", is(notNullValue()));
```
Comment (farukgoc) [inline src/test/java/com/library/steps/B28G31_US190_StepDefs_NC.java]: I dont think you need an if condition here. Just replace the hardcoded "id" with field. That will work for both "id" and "name" cases

**Review by farukgoc — CHANGES_REQUESTED** [review]
I tested your codes and it is working fine. I still added some comments for better logic and cleaner code. You might wanna check Alev's code for a reference. It is up to you if you wanna make changes because your code works tis way too

File: src/test/java/com/library/steps/B28G31_US190_StepDefs_NC.java (hunk 4 above)
Comment (nurseda916) [reply src/test/java/com/library/steps/B28G31_US190_StepDefs_NC.java]: actually that's a good idea i'll try that way, thanks.

**Review by farukgoc — APPROVED** [review]
Thank you very much Nurseda. Good job.

**Review by untuzina — APPROVED** [review]
Good Job!

**Review by JavBobby — APPROVED** [review]

**Pull request merged**